Dockerfile

```diff
@@ -0,0 +1,81 @@
+FROM ubuntu:bionic-20220427 as builder
+
+WORKDIR /work
+
+RUN apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y \
+	    build-essential \
+	    pkg-config \
+	    checkinstall \
+	    git \
+	    autoconf \
+	    automake \
+	    libtool-bin \
+	    libplist-dev \
+        libavahi-glib-dev libavahi-client-dev \
+        libimobiledevice-dev \
+        libusb-1.0-0-dev \
+        libssl-dev \
+        udev \
+        libplist++-dev libtool autoconf automake \
+        python3 python3-dev \
+        curl usbmuxd \
+        wget lsb-release wget software-properties-common \
+        clang-10
+
+RUN for i in /etc/ssl/certs/*.pem; do HASH=$(openssl x509 -hash -noout -in $i); ln -s $(basename $i) /etc/ssl/certs/$HASH.0 || true; done
+
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+
+RUN git clone https://github.com/jkcoxson/rusty_libimobiledevice.git \
+    && git clone https://github.com/jkcoxson/plist_plus.git \
+    && git clone https://github.com/libimobiledevice/libimobiledevice-glue.git \
+    && git clone https://github.com/zeyugao/zeroconf-rs.git \
+    && git clone https://github.com/libimobiledevice/libplist.git \
+    && git clone https://github.com/libimobiledevice/libusbmuxd.git \
+    && git clone https://github.com/jkcoxson/mdns.git
+
+RUN cd libplist \
```

Comment:
I don't think cloning these are necessary, as [it does the same thing here](https://github.com/jkcoxson/rusty_libimobiledevice/blob/f8fd18f39c74d821258192a26b4e0c930fb48c85/build.rs#L48). That will add to the build time significantly if we're building twice.

Reply:
I rechecked, if I remove `libplist`, `libimobiledevice-glue` and `libusbmuxd`, it will be not able to compile.

Reply:
I think that in vendored mode, the configuration provided by the `build.rs` will only affect the compilation itself? Not the global compliation.

Reply:
The only dependency that need libplist etc is rusty_libimobiledevice, and it tells cargo to link it itself.

Reply:
Maybe I found out why. When building `libimobiledevice` in vendored mode, it requires libusbmuxd. But you didn't add the built libusbmuxd path to libimobiledevice https://github.com/jkcoxson/rusty_libimobiledevice/blob/f8fd18f39c74d821258192a26b4e0c930fb48c85/build.rs#L161
As a result, I add the libusbmuxd as dep, recursively libimobiledevice-glue and libplist

Reply:
I see what you mean. I'll have to update the build system for libimobiledevice, but until then we should have a feature to disable a vendored rusty_libimobiledevice build.

Reply:
I found a [typo](https://github.com/jkcoxson/rusty_libimobiledevice/commit/552eadf0c07ad9a0d86cb76d973791798918c77e) in my build script, it may fix the issue

Reply:
```
#18 43.00   configure: error: Package requirements (libplist-2.0 >= 2.2.0) were not met:
#18 43.00
#18 43.00   No package 'libplist-2.0' found
#18 43.00
#18 43.00   Consider adjusting the PKG_CONFIG_PATH environment variable if you
#18 43.00   installed software in a non-standard prefix.
#18 43.00
#18 43.00   Alternatively, you may set the environment variables libplist_CFLAGS
#18 43.00   and libplist_LIBS to avoid the need to call pkg-config.
#18 43.00   See the pkg-config man page for more details.
#18 43.00   thread 'main' panicked at '
#18 43.00   command did not execute successfully, got: exit status: 1
```

As the error message suggest, we should change the `PKG_CONFIG_PATH` env?

Reply:
This error is fixed in [this commit](https://github.com/jkcoxson/rusty_libimobiledevice/commit/eddfea5b3aab76606f84f09de30d5066f72ae9ab)

Reply:
Dumped commit